src/bin/cli.js

```diff
@@ -6,138 +6,155 @@ const chalk = require('chalk');
-      } else {
-        handlingContext.ready = true;
-        handlingContext.file = filepath;
-        if (!handlingContext.scenarioFile) {
-          inputLoopScenario(handlingContext);
-        }
-        // eslint-disable-next-line sonarjs/no-identical-functions
-        filesystem.access(handlingContext.scenarioFile, 1, (err) => {
-          if (err) {
-            console.log(`\nError in accessing provided ${chalk.blueBright('scenario')} file \nDetails:${err}\n\n`);
-            inputLoopScenario(handlingContext);
-          } else if (!String(handlingContext.scenarioFile).match(handlingContext.yamlJsonRegex)) {
-            console.log(`\nPlease provide a proper ${chalk.blueBright('scenario')} filepath ex: ./myAsyncApi.json ./myAsyncApi.yaml:`);
-            inputLoopScenario(handlingContext);
+const inputLoopScenario = (rd,scenario,regex) => {
+  const res = checkFilepath(scenario,regex);
+
+  if (!res) {
+    return new Promise((resolve) => {
+      rd.question('\nPlease provide an existent yaml or json file .It should abide by the scenario json schema.\nScenario filepath:',(answer) => {
+        const inputLoop = (filepath) => {
+          const res = checkFilepath(filepath,regex);
+          if (!res) {
+            rd.question('Please fix errors and provide a correctly formatted and accessible file in filepath.\nScenario filepath:',(answer) => {
+              inputLoop(answer);
+            });
           } else {
-            //handlingContext.scenarioFile = handlingContext.scenarioFile;
-            handlingContext.scenarioReady = true;
-            parseAsyncApi(handlingContext);
+            resolve(filepath);
           }
-        }
-        );
-      }
+        };
+
+        inputLoop(answer);
+      });
     });
+  }
+  return  new Promise((resolve) => {resolve(scenario);});
+};
+
+const inputLoopAsyncApi = (rd,asyncFile,regex) => {
+  const res = checkFilepath(asyncFile,regex);
+
+  if (!res) {
+    return new Promise((resolve) => {
+      rd.question('\nPlease provide an existent yaml or json file.It should abide by the asyncApi Spec.\nAsyncApi filepath:',(answer) => {
+        const inputLoop = (filepath) => {
+          const res = checkFilepath(filepath,regex);
+          if (!res) {
+            rd.question('Please fix errors and provide a correctly formatted and accessible file in filepath.\nAsyncApi Filepath:',(answer) => {
+              inputLoop(answer);
+            });
+          } else resolve(filepath);
+        };
+
+        inputLoop(answer);
+      });
+    });
+  }
+  return new Promise((resolve) => {
+    resolve(asyncFile);
   });
 };
 
 /**
- * Verifies the command line arguments, re-prompts in case of error. Parses file and returns the object representation.
+ * Verifies the command line arguments, re-prompts in case of error. Parses asyncApiF and returns the object representation.
  * @returns {Promise<*|null>}
  * @param rd
- * @param file
+ * @param asyncApiF
```

Comment:
I dont think `asyncApiF` makes sense 😅 What about being precise and call it `asyncApiFilePath`?